### DISCUSSION TITLE
MongoDB: createMany with `skipDuplicates` flag

### DISCUSSION THREAD

**Discussion by kraikov** [prisma/prisma#16115] [Q&A] [answered]
Is it possible inserting multiple documents without aborting if there is a duplicate?

The way this can be achieve in mongodb is by using `insertMany` with `ordered` flag to `false`

Is there a way to achieve the same with Prisma?

Example:

```ts
db.collection.insertMany(
   [ <document 1> , <document 2>, ... ],
   {
     ordered: false
   }
)
```

**Comment by pantharshit00** [comment, marked as answer]
Hey @kraikov 

Currently it is not possible with MongoDB as we do `ON CONFLICT DO NOTHING` for the SQL drivers. You can open a feature request for this here: https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=

As a workaround, you might be able to use the raw query API here: https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#raw-queries-with-mongodb

**Reply by inter-gobhr** [reply]
I would request you to please update in documentation in [](https://www.prisma.io/docs/concepts/components/prisma-client/crud) as a note **skipDuplicate** is not supported with MongoDB , It can save developer time . In our multiple project we user **prisma** everywhere , Hence I am waiting for this feature. 
Thanks

**Reply by janpio** [reply]
Where do you think we should add that? Sounds useful.

**Reply by lperrious** [reply]
For instance, it would have been useful for me today here : https://www.prisma.io/docs/concepts/components/prisma-client/crud#create

**Reply by inter-gobhr** [reply]
https://www.prisma.io/docs/concepts/components/prisma-client/crud#create-multiple-records
Create multiple records
Prisma Client supports bulk inserts as a GA feature in [2.20.0](https://github.com/prisma/prisma/releases/2.20.0) and later.

The following [createMany](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#createmany)  query creates multiple users and skips any duplicates (email must be unique):

```
const createMany = await prisma.user.createMany({
  data: [
    { name: 'Bob', email: 'bob@prisma.io' },
    { name: 'Bobo', email: 'bob@prisma.io' }, // Duplicate unique key!
    { name: 'Yewande', email: 'yewande@prisma.io' },
    { name: 'Angelique', email: 'angelique@prisma.io' },
  ],
  skipDuplicates: true, // Skip 'Bobo'
 **// skipDuplicates available in only relation database .**
})
```
Or somewhere below you can put note : **skipDuplicates** flag will work only with relational  database .

**Reply by janpio** [reply]
That is very fair. I raised this internally. Thanks for pointing it out!

**Reply by sabinadams** [reply]
This is being addressed here: https://github.com/prisma/docs/issues/4593
Thanks for the feedback!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!

**Comment by kulterryan** [comment]
Any updates on this thread?
